packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -74,24 +103,28 @@ class TransferEngine implements ITransferEngine {
   }
 
   async integrityCheck(): Promise<boolean> {
-    const sourceMetadata = await this.sourceProvider.getMetadata();
-    const destinationMetadata = await this.destinationProvider.getMetadata();
+    try {
+      const sourceMetadata = await this.sourceProvider.getMetadata();
+      const destinationMetadata = await this.destinationProvider.getMetadata();
 
-    if (!sourceMetadata || !destinationMetadata) {
-      return true;
-    }
+      if (sourceMetadata && destinationMetadata) {
+        this.#assertStrapiVersionIntegrity(
+          sourceMetadata?.strapi?.version,
+          destinationMetadata?.strapi?.version
+        );
+      }
 
-    try {
-      // Version check
-      this.assertStrapiVersionIntegrity(
-        sourceMetadata?.strapi?.version,
-        destinationMetadata?.strapi?.version
-      );
+      const sourceSchemas = await this.sourceProvider.getSchemas?.();
+      const destinationSchemas = await this.destinationProvider.getSchemas?.();
+
+      if (sourceSchemas && destinationSchemas) {
+        this.#assertSchemasMatching(sourceSchemas, destinationSchemas);
+      }
 
       return true;
     } catch (error) {
       if (error instanceof Error) {
-        console.error('Integrity checks failed:', error.message);
```

Comment:
I think we can keep `error.message` here, I just removed it when we were debugging where the error was coming from but I'm not sure it makes sense to keep the stack trace in the log.